File: dotnet/src/Microsoft.Agents.Protocol.Validation/Validators/ContentValidators/AdaptiveCardContentValidator.cs
using Microsoft.Agents;

namespace Microsoft.Agents.Protocol.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for AdaptiveCardContent.
/// </summary>
public class AdaptiveCardContentValidator : ContentValidatorBase<AdaptiveCardContent>
{
    public override ValidationResult Validate(AdaptiveCardContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-014: AdaptiveCardContent card must be valid JSON
        var jsonError = ValidateJson(content.Card, "Card",
            ValidationErrorCode.CNT_014,
            "AdaptiveCardContent card must be valid JSON");
        if (jsonError != null)
            errors.Add(jsonError);

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/AdaptiveCardContentValidator.ts
import type { AdaptiveCardContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';

/**
 * Validator for AdaptiveCardContent.
 */
export class AdaptiveCardContentValidator extends ContentValidatorBase<AdaptiveCardContent> {
  public validate(content: AdaptiveCardContent, context?: ValidationContext): ValidationResult {
    // AdaptiveCardContent has no strict validation requirements
    return ValidationResult.success();
  }
}
